setup.py

```diff
@@ -589,284 +596,300 @@ def get_blosc_version(headername):
+                _platdep["HDF5"] = [hdf5_old_dll_name, hdf5_old_dll_name]
                 _, libdir, rundir = package.find_directories(
-                    location, use_pkgconfig=USE_PKGCONFIG)
+                    location, use_pkgconfig=USE_PKGCONFIG
+                )
 
         # check if the library is in the standard compiler paths
         if not libdir and package.target_function:
-            libdir = compiler.has_function(package.target_function,
-                                        libraries=(package.library_name,))
+            libdir = compiler.has_function(
+                package.target_function, libraries=(package.library_name,)
+            )
 
         if not (hdrdir and libdir):
-            if package.tag in ['HDF5']:  # these are compulsory!
+            if package.tag in ["HDF5"]:  # these are compulsory!
                 pname, ptag = package.name, package.tag
                 exit_with_error(
-                    "Could not find a local %s installation." % pname,
-                    "You may need to explicitly state "
-                    "where your local %(name)s headers and library can be found "
-                    "by setting the ``%(tag)s_DIR`` environment variable "
-                    "or by using the ``--%(ltag)s`` command-line option."
-                    % dict(name=pname, tag=ptag, ltag=ptag.lower()))
-            if package.tag == 'BLOSC':  # this is optional, but comes with sources
-                print("* Could not find %s headers and library; "
-                    "using internal sources." % package.name)
+                    f"Could not find a local {pname} installation.",
+                    f"You may need to explicitly state where your local "
+                    f"{pname} headers and library can be found by setting "
+                    f"the ``{ptag}_DIR`` environment variable or by using "
+                    f"the ``--{ptag.lower()}`` command-line option.",
+                )
+            if package.tag == "BLOSC":
+                # this is optional, but comes with sources
+                print(
+                    f"* Could not find {package.name} headers and library; "
+                    f"using internal sources."
+                )
             else:
-                print("* Could not find %s headers and library; "
-                    "disabling support for it." % package.name)
+                print(
+                    f"* Could not find {package.name} headers and library; "
+                    f"disabling support for it."
+                )
 
             continue  # look for the next library
 
         if libdir in ("", True):
-            print("* Found %s headers at ``%s``, the library is located in the "
-                "standard system search dirs." % (package.name, hdrdir))
+            print(
+                f"* Found {package.name} headers at ``{hdrdir}``, the library "
+                f"is located in the standard system search dirs."
+            )
```

Comment:
OK ``True`` means that the compiler found the library somewhere in the system.

Reply:
So what do we do in the example above? `if hdrdir` with `True/str` is true if `True` or the `str` is not empty. But if it is only `True`, we cannot `os.path.join` it to some other string. What are the defaults? Is that `True` actually necessary? Can we do it a `Path` if found, `None` if not?

Reply:
Sorry I don't remember all details.
Yes I'm sure that there are better alternatives to the current implementation.
The point here is that.there are some cases in which we know the a library is available in the system but we do not know its path.
For this specific case ``None`` do not helps, we need some other kind of mechanism.

Reply:
Postponed to a new PR.